### PR TITLE
Staging

### PR DESCRIPTION
**Pull request by westalert**
merge request from staging to master

### CODE REVIEW

File: src/components/InfinityTable/InfinityTable.tsx [hunk 1]
```diff
@@ -102,7 +108,11 @@ const InfinityTableComponent: React.FC<IInfinityTableComponentProps> = ({
   renderAllRows,
 }) => {
   const [loading, setLoading] = React.useState(false);
-
+  const isDarkMode = useGetThemeMode();
+  const [color, bgOpacity] = React.useMemo(
+    () => (isDarkMode ? ['#232630', 0.5] : ['#FFFFFF', 0.05]),
```
Comment (CrazyOptimist) [inline src/components/InfinityTable/InfinityTable.tsx:113]: Please use colors from theme, as hardcoding colors looks not very good

File: src/components/SearchBar/SwitchMode.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    dispatch(setAppThemeAction(value));
+    localStorage.setItem('darkMode', value ? 'true' : 'false');
+  }, []);
+
+  return (
+    <div>
+      <FormControlLabel
+        control={<Switch checked={isDarkMode} onChange={handleChangeMode} />}
+        label={isDarkMode ? 'Light' : 'Dark'}
+        style={{ color: isDarkMode ? '#F8F8FA' : '#2D3748' }}
```
Comment (CrazyOptimist) [inline src/components/SearchBar/SwitchMode.tsx:22]: Use colors from theme

File: src/components/Sidebar/Sidebar.tsx [hunk 3]
```diff
@@ -179,14 +180,19 @@ const Sidebar: React.FC<RouteComponentProps & SidebarPropsType> = ({ location, .
...
+            </Hidden>
           </Styles.Items>
         </List>
       </Styles.Scrollbar>
       <Styles.SidebarFooter>
         <Styles.SidebarContainer container spacing={2} justify="space-around" alignItems="center">
           {footerIcons.map(({ id, url, icon }) => (
             <Grid item key={id}>
-              <IconButton target="_blank" href={url} color="primary" style={{ padding: '4px' }}>
+              <IconButton target="_blank" href={url} style={{ color: '#6151d0', padding: '4px' }}>
```
Comment (CrazyOptimist) [inline src/components/Sidebar/Sidebar.tsx:195]: If it's complex to use color from theme, consider defining and using css variables, they can be changed by dark mode as well.
But I guess theme should be enough for managing colors, isn't it?

File: src/pages/HistoricalStatistics/BlockchainSize/index.tsx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  const [currentBgColor, handleBgColorChange] = useBackgroundChart();
+  const [period, setPeriod] = useState<PeriodTypes>(CHART_DEFAULT_PERIOD);
+  const fetchStats = useFetch<{ data: Array<TTransactionsChart> }>({
+    method: 'get',
+    url: URLS.GET_BLOCKS_CHARTS,
+  });
+  useEffect(() => {
+    const loadLineChartData = async () => {
+      const data = await fetchStats.fetchData({
+        params: { period, sortDirection: 'DESC', sqlQuery: 'SUM(size)' },
```
Comment (CrazyOptimist) [inline src/pages/HistoricalStatistics/BlockchainSize/index.tsx:26]: `sqlQuery` here should be changed to not pass sql directly. Maybe such parameter will fit: `aggregate: 'sumSize'`

File: src/pages/HistoricalStatistics/Difficulty/index.tsx [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+import { EChartsLineChart } from '../Chart/EChartsLineChart';
+
+function Difficulty() {
+  const [chartData, setChartData] = useState<TLineChartData | null>(null);
+  const [currentBgColor, handleBgColorChange] = useBackgroundChart();
+  const [period, setPeriod] = useState<PeriodTypes>(CHART_DEFAULT_PERIOD);
+  const fetchStats = useFetch<{ data: Array<IStatistic> }>({
+    method: 'get',
+    url: URLS.GET_STATISTICS,
+  });
```
Comment (CrazyOptimist) [inline src/pages/HistoricalStatistics/Difficulty/index.tsx:22]: I can see no loading and no error handling here,
I suggest to use `react-query`: it takes care of caching, provide loading and error state.

**Review by CrazyOptimist — CHANGES_REQUESTED** [review]

**Review by CrazyOptimist — DISMISSED** [review]

**Review by Dicklesworthstone — DISMISSED** [review]

**Review by CrazyOptimist — APPROVED** [review]

**Review by Dicklesworthstone — APPROVED** [review]

**Pull request merged**